contracts/wormhole/contracts/mapping-token/v2/Erc1155MappingTokenFactory.sol

```diff
@@ -0,0 +1,180 @@
+     * @param bridgedLanePosition the bridged lane position to send the unlock message
+     * @param mappingToken the burt mapping token address
+     * @param recipient the recipient of the remote unlocked token
+     * @param ids the ids of the burn and unlock
+     */
+    function burnAndRemoteUnlockWaitingConfirm(
+        uint32 bridgedLanePosition,
+        address mappingToken,
+        address recipient,
+        uint256[] memory ids,
+        uint256[] memory amounts
+    ) external payable whenNotPaused {
+        require(ids.length > 0, "Erc1155MappingTokenFactory:can not transfer empty id");
+        address originalToken = mappingToken2OriginalToken[mappingToken];
+        require(originalToken != address(0), "Erc1155MappingTokenFactory:token is not created by factory");
+        // Lock the fund in this before message on remote backing chain get dispatched successfully and burn finally
+        // If remote backing chain unlock the origin token successfully, then this fund will be burned.
+        // Otherwise, these tokens will be transfered back to the msg.sender.
+        bytes[] memory attrs = new bytes[](ids.length);
+        address serializer = IErc1155MappingToken(mappingToken).attributeSerializer();
+        IERC1155(mappingToken).safeBatchTransferFrom(msg.sender, address(this), ids, amounts, "");
+        for (uint256 idx = 0; idx < ids.length; idx++) {
+            if (serializer != address(0)) {
+                attrs[idx] = IErc1155AttrSerializer(serializer).serialize(ids[idx]);
+            }
+        }
+
+        bytes memory unlockFromRemote = abi.encodeWithSelector(
+            IErc1155Backing.unlockFromRemote.selector,
+            address(this),
+            originalToken,
+            recipient,
+            ids,
+            amounts,
+            attrs
+        );
+        uint256 messageId = _sendMessage(bridgedLanePosition, unlockFromRemote);
+        unlockRemoteUnconfirmed[messageId] = UnconfirmedInfo(msg.sender, mappingToken, ids, amounts);
+        emit BurnAndWaitingConfirm(messageId, msg.sender, recipient, mappingToken, ids, amounts);
+    }
+
+    /**
+     * @notice this will be called when the burn and unlock from remote message confirmed
+     * @param messageId the message id, is used to identify the unlocked message
+     * @param result the result of the remote backing's unlock, if false, the mapping token need to transfer back to the user, otherwise burt
+     */
+    function on_messages_delivered(
+        uint256 messageId,
+        bool result
+    ) external onlyOutBoundLane {
+        UnconfirmedInfo memory info = unlockRemoteUnconfirmed[messageId];
+        require(info.ids.length > 0 && info.sender != address(0) && info.mappingToken != address(0), "Erc1155MappingTokenFactory:invalid unconfirmed message");
+        if (result) {
+            IErc1155MappingToken(info.mappingToken).burnBatch(info.ids, info.amounts);
```

Comment:
After burn the tokens, May the `serializer` need to clean the `Attributes` ?

Reply:
After disscussion, we think we should not serialize/deserialize attributes while transfer the erc1155 tokens. The attributes belong to the class of the tokens which have the same tokenId.